File: front-end/src/application/usecase/workflow/workflow-get-use-case.ts
import { WorkFlow } from '@/domain/entity/workflow';
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';

export interface WorkFlowGetUseCase {

    /**
     * Get a workflow by its ID
     * 
     * @param workflowId - The ID of the workflow
     * @returns The workflow if found, otherwise undefined
     */
    getWorkFlow(workflowId: UllUUID): Promise<Either<DataError, WorkFlow>>;

    /**
     * Get all flows
     * 
     * @returns An array of all flows
     */
    getAllWorkFlows(): Promise<Either<DataError, WorkFlow[]>>;
}

File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowGetUseCase.java
package es.ull.project.application.usecase.workflow;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import es.ull.project.domain.entity.WorkFlow;

public interface WorkFlowGetUseCase {

    /**
     * Get an workflow by ID
     * 
     * @param workflow
     * @return workflow selected
     */
    Optional<WorkFlow> getWorkFlow(UUID workflow);

    /**
     * Get all flows
     * 
     * @return all flows
     */
    List<WorkFlow> getAllWorkFlows();
}
